pkg/virt-controller/watch/migration_test.go

```diff
@@ -49,84 +53,368 @@ var _ = Describe("Migration", func() {
 			&inject.Object{Value: templateService},
 		)
 		g.Populate()
-
-		stopChan = make(chan struct{})
-		lw = framework.NewFakeControllerSource()
 		migrationCache = cache.NewIndexer(cache.DeletionHandlingMetaNamespaceKeyFunc, nil)
+		dispatch = NewMigrationControllerDispatch(vmService)
+		queue = workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+		// Create a VM which is being scheduled
+
+		vm = v1.NewMinimalVM("testvm")
+		vm.Status.Phase = v1.Running
+		vm.ObjectMeta.SetUID(uuid.NewUUID())
+
+		migration = v1.NewMinimalMigration(vm.ObjectMeta.Name+"-migration", vm.ObjectMeta.Name)
+		migration.ObjectMeta.SetUID(uuid.NewUUID())
+		migration.Spec.NodeSelector = map[string]string{"beta.kubernetes.io/arch": "amd64"}
 
-		_, migrationController = NewMigrationControllerWithListWatch(vmService, nil, lw)
+		// Create a target Pod for the VM
+		templateService, err := services.NewTemplateService("whatever")
+		Expect(err).ToNot(HaveOccurred())
+		pod, err = templateService.RenderLaunchManifest(vm)
+		Expect(err).ToNot(HaveOccurred())
+
+		pod.Spec.NodeName = "mynode"
+		pod.Status.Phase = clientv1.PodSucceeded
+		pod.Labels[v1.DomainLabel] = migration.ObjectMeta.Name
+
+		podList = clientv1.PodList{}
+		podList.Items = []clientv1.Pod{*pod}
 
-		// Start the controller
-		migrationController.StartInformer(stopChan)
-		go migrationController.Run(1, stopChan)
 	})
 
+	doExecute := func() {
+		key, _ := cache.MetaNamespaceKeyFunc(migration)
+		queue.Add(key)
+		dispatch.Execute(migrationCache, queue, key)
+
+	}
+
+	buildExpectedVM := func(phase v1.VMPhase) *v1.VM {
+
+		obj, err := conversion.NewCloner().DeepCopy(vm)
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedVM := obj.(*v1.VM)
+		expectedVM.Status.Phase = phase
+		expectedVM.Status.MigrationNodeName = pod.Spec.NodeName
+		expectedVM.Spec.NodeSelector = map[string]string{"beta.kubernetes.io/arch": "amd64"}
+
+		return expectedVM
+	}
+
+	handlePutMigration := func(migration *v1.Migration, expectedStatus v1.MigrationPhase) http.HandlerFunc {
+
+		obj, err := conversion.NewCloner().DeepCopy(migration)
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedMigration := obj.(*v1.Migration)
+		expectedMigration.Status.Phase = expectedStatus
+		expectedMigration.Kind = "Migration"
+		expectedMigration.APIVersion = "kubevirt.io/v1alpha1"
```

Comment:
Setting all these fields looks strange. I think it is better if you only explicitly set fields which are relevant for the test.

Self link, ... are not something we  need to check here ...